aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -31,6 +31,122 @@
 namespace at {
 namespace native {
 
+namespace {
+  enum class TypePromotionStrategy {
```

Comment:
I'm hoping we can simplify this, but if we need the whole Enum we probably need more descriptive names vs `TypeN`

Reply:
Thanks for your inputs @nairbv. We went with the `Enum` approach, as the unary ops have different dtype promotion policies, and we needed an easier way for each op to communicate with the dtype promotion policy. We also wanted to not pollute the helper function `unary_op_impl` with any type promotion strategies, and to keep it short, we use helper functions defined in the anonymous namespace.

I agree, that the names need to be more descriptive. The next commit will resolve that.

Reply:
I've changed the names to more descriptive names in the Enum. (https://github.com/pytorch/pytorch/pull/33322/commits/2203eb9fae716bad2b5907dfc4518323ca4ee5c5)